python/lsst/ts/m2gui/controltab/tab_rigid_body_pos.py

```diff
@@ -33,7 +37,326 @@ class TabRigidBodyPos(TabDefault):
+    ----------
+    model : `Model`
+        Model class.
     """
 
+    # The following two maximum values come from the ts_mtm2 that originated
+    # by vendor. Need to figure out the details in a latter time.
+
+    # Maximum movement in x, y, or z direction in millimeter
+    MAX_DISTANCE_IN_MM = 4
+
+    # Maximum rotation in rx, ry, or rz direction in microradian
+    MAX_ROTATION_IN_URAD = 20000
+
     def __init__(self, title, model):
         super().__init__(title, model)
+
+        # Relative target position
+        self._target_position_relative = self._set_target_position()
+
+        # Absolute target position
+        self._target_position_absolute = self._set_target_position()
+
+        # Buttons in this table
+        self._buttons = {
+            "save_position": set_button("Save Position", self._callback_save_position),
+            "go_to_home": set_button("Go to Home", self._callback_go_to_home),
+            "set_home": set_button("Set Home", self._callback_set_home),
+            "jog": set_button("Jog", self._callback_jog),
+            "clear_values_relative": set_button(
+                "Clear Values",
+                self._callback_clear_values,
+                self._target_position_relative,
+            ),
+            "go_to_position": set_button(
+                "Go to Position", self._callback_go_to_position
+            ),
+            "clear_values_absolute": set_button(
+                "Clear Values",
+                self._callback_clear_values,
+                self._target_position_absolute,
+            ),
+        }
+
+        # Rigid body position
+        self._position = self._set_position()
+
+        # Internal layout
+        self.widget().setLayout(self._create_layout())
+
+        self._set_signal_position(self.model.utility_monitor.signal_position)
+
+    def _set_target_position(self):
+        """Set the target position of rigid body.
+
+        Returns
+        -------
+        position : `dict`
+            Target position of the rigid body. The key is the axis. The value
+            is the `PySide2.QtWidgets.QDoubleSpinBox`.
+        """
+
+        num_digit_after_decimal = (
+            self.model.utility_monitor.NUM_DIGIT_AFTER_DECIMAL_DISPLACEMENT
+        )
+
+        position = dict()
+        for axis in self._get_axes():
+            double_spin_box = QDoubleSpinBox()
+            double_spin_box.setDecimals(num_digit_after_decimal)
+            double_spin_box.setSingleStep(get_tol(num_digit_after_decimal))
+
+            if axis.startswith("r"):
+                double_spin_box.setRange(
+                    -self.MAX_ROTATION_IN_URAD, self.MAX_ROTATION_IN_URAD
+                )
+            else:
+                double_spin_box.setRange(
+                    -self.MAX_DISTANCE_IN_MM, self.MAX_DISTANCE_IN_MM
+                )
+
+            position[axis] = double_spin_box
+
+        return position
+
+    def _get_axes(self):
+        """Get the axes of position.
```

Comment:
This doesn't have to be a method, as this isn't supposed to change. AXES = ["..] in class definition and then calling self.AXES shall work and will save few CPU cycles.

Reply:
Fixed in the commit: [fdbacbb](https://github.com/lsst-ts/ts_m2gui/pull/9/commits/fdbacbbe2fa461e38889c8d7a022b58f6008c86c). Thanks!